simple/lint.go

```diff
@@ -68,6 +68,43 @@ func (c *Checker) Funcs() map[string]lint.Func {
 		"S1028": c.LintErrorsNewSprintf,
 		"S1029": c.LintRangeStringRunes,
 		"S1030": c.LintBytesBufferConversions,
+		"S1031": c.LintNilCheckAroundRange,
+	}
+}
+
+func (c *Checker) LintNilCheckAroundRange(j *lint.Job) {
```

Comment:
Forget my silly comment regarding nil maps. Ranging over a nil map will never execute the loop body. So, we should check that the type is not a channel, slices and maps are fine.